contracts/core/Registry.sol

```diff
@@ -1,47 +1,251 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-// use Solady Roles for ownership of projects: https://github.com/Vectorized/solady/blob/main/src/auth/OwnableRoles.sol
-contract Registry {
+import {AccessControl} from "../../lib/openzeppelin-contracts/contracts/access/AccessControl.sol";
+
+import {Metadata} from "./libraries/Metadata.sol";
+
+contract Registry is AccessControl {
+    error NO_ACCESS_TO_ROLE();
+    error NOT_SUPPORTED();
+
+    /// @notice Unique identifier for an identity - a counter
+    uint private identityId;
+
+    /// @notice Types of roles assigned to an identity
+    enum RoleType {
+        OWNER,
+        MEMBER
+    }
+
+    /// @notice Struct to hold details of an identity
     struct IdentityDetails {
         string name;
-        string metadata; // @todo update to MetaPtr
-        address attestationAddr;
+        Metadata metadata;
+        address anchor;
     }
 
-    // getter for projects mapping
-    function identities(uint _identityId) external view returns (IdentityDetails memory) {
+    /// @notice identityId -> IdentityDetails
+    mapping(uint => IdentityDetails) public identities;
+
+    // Events
+    event IdentityCreated(
+        uint indexed identityId,
+        string name,
+        Metadata metadata,
+        address anchor
+    );
+    event IdentityNameUpdated(
+        uint indexed identityId,
+        string name,
+        address newOwnerIdentifier
+    );
+    event IdentityMetadataUpdated(uint indexed identityId, Metadata metadata);
+
+    /// @notice Creates a new identity
+    /// @dev This will also set the attestation address generated from msg.sender and name
+    /// @param name The name of the identity
+    /// @param metadata The metadata of the identity
+    /// @param _owners The owners of the identity
+    /// @param _members The members of the identity
+    function createIdentity(
+        string memory name,
+        Metadata memory metadata,
+        address[] memory _owners,
```

Comment:
Can we standardize how we write the params, I am fine with either way, i.e.  `_param, param`. Typically I use without _ and only use that for private/internal variables and functions.

Reply:
I think the most common style is to use the underscore at the beginning of a function parameter (i.e. `_param`) and otherwise to not use them. That would be my preference

Reply:
let's use prefix underscore